### PR TITLE
PP 392 add error screen events

### PR DESCRIPTION
**Pull request by ValentinaIancu-Gini**
- events added: screen_shown, close_tapped, enter_manually_tapped, retake_images_tapped
- refactor no results screen screen_shown event properties to use AnalyticsMapper

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift [hunk 1]
```diff
@@ -139,4 +148,15 @@ import GiniBankAPILibrary
                 comment: "Outage error")
         }
     }
+
+    /**
+     Get the error analytics for the current `ErrorType`.
+
+     - Returns: An `ErrorAnalytics` object representing the analytics for the error.
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift:155]: [Clarification]: We've discussed it already and short summary: 
 - Function is a mapping a GiniError into ErrorAnalytics
 -  it returns an error in special format for the analytics
 Answer from @ValentinaIancu-Gini : Returns: An `ErrorAnalytics` object representing the error for the analytics.

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Great job @ValentinaIancu-Gini 
just 2 tiny suggestions from my side

File: BankAPILibrary/GiniBankAPILibrary/Sources/GiniBankAPILibrary/Documents/Core/GiniError.swift [hunk 2]
```diff
@@ -48,9 +48,9 @@ public enum GiniError: Error, GiniErrorProtocol, Equatable {
     case requestCancelled
     case tooManyRequests(response: HTTPURLResponse? = nil, data: Data? = nil)
     case unauthorized(response: HTTPURLResponse? = nil, data: Data? = nil)
-    case maintenance
-    case outage
-    case server
+    case maintenance(errorCode: Int)
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankAPILibrary/GiniBankAPILibrary/Sources/GiniBankAPILibrary/Documents/Core/GiniError.swift:51]: An HTTP response status code can not be negative. 
 https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift (hunk 1 above)
Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift:155]: I've added more comments and edit this one.

File: BankAPILibrary/GiniBankAPILibrary/Sources/GiniBankAPILibrary/Documents/Core/GiniError.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankAPILibrary/GiniBankAPILibrary/Sources/GiniBankAPILibrary/Documents/Core/GiniError.swift:51]: Also `    statusCode: Int,` in 
```
 private func handleError<T: Resource>(resource: T,
                                          statusCode: Int,
                                          response: HTTPURLResponse?,
                                          data: Data?,
                                          taskType: TaskType,
                                          cancellationToken: CancellationToken?,
                                          completion: @escaping CompletionResult<T.ResponseType>) {
```

**Review by zladzeyka — APPROVED** [review]
@ValentinaIancu-Gini  Thank you!

**Pull request merged**